core/src/main/java/org/infinispan/commands/functional/ReadOnlyKeyCommand.java

```diff
@@ -42,6 +66,7 @@ public void writeTo(ObjectOutput output) throws IOException {
       output.writeObject(key);
       output.writeObject(f);
       Params.writeObject(output, params);
+      output.writeObject(encodingClasses);
```

Comment:
@rvansa I said that in the description, @gustavonalle is working on a PR whose refactoring will be the one. My encoding classes are more something I did here because passing 4 classes every time was too heavy ; specially knowing that I will have to change all that after Gustavo